dev/py-validate.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
```

Comment:
@davies what do you think about including this vs just enforcing the checks? If it would be used, I'm for it, but not clear how often this would be run in practice by anyone if we're not going to use its big-bang changes.